monday/query_joins.py

```diff
@@ -289,12 +289,28 @@ def get_tags_query(tags):
 
 
 # BOARD RESOURCE QUERIES
-def get_board_items_query(board_id):
+def get_board_items_query(board_id: Union[str, int], limit: Optional[int] = None, page: Optional[int] = None) -> str:
+    def _gather_params(params) -> List[str]:
```

Comment:
While it's true that this method is only used here at the moment, I can see this being a useful util to reuse in the future with some tweaks (i.e. remove board specific logic - thoughts on passing in the list of invalid params, `board_id` and `_gather_params`, in here to make this generic?). I'd prefer that this method live outside here, in `utils.py`

Reply:
ya, that's kinda why I like this style of nested functions. Makes it obvious what will need to be pulled out as common patterns start to appear. I can do another pass and get this into util